### PR TITLE
feat: Add ability to fire all secondary weapons

### PR DESCRIPTION
**Pull request by petervdmeer**
**Feature:** This PR implements a feature related to this [comment](https://github.com/endless-sky/endless-sky/issues/1724#issuecomment-254778984)

## Feature Details
This PR adds an "all secondary weapons selected" option after the currently available "no secondary weapon selected" and "single secondary weapon selected" options.
The internal implementation uses a set which will also be suitable for the "mouse selectable set of options" as discussed in https://github.com/endless-sky/endless-sky/issues/1724.

## UI Screenshots
Screenshot of all weapons selected:
all_weapons_selected

## Usage Examples
Just cycle through the list of available secondary weapons and notice that the new all-weapons-selected option is now also available at the end of the list.

## Testing Done
Tested with firing all secondary weapons (on a ship with 3 separate secondary weapons installed).
Tested with firing individual secondary weapons (on a ship with 3 separate secondary weapons installed).
Tested the selection and firing of a single weapon on a single ship.

## Performance Impact
I don't expect any performance impact (the player keys are checked once per frame only).

### CODE REVIEW

**Comment by tehhowch** [comment]
I'd still like that option present.

Default to none, then individual, then all.

Similarly, the user should be able to do Shift / Ctrl clicks with the selected secondaries, to e.g. select the first and third, or deselect one, etc.

**Comment by petervdmeer** [comment]
> I'd still like that option present.
>
> Default to none, then individual, then all.

Ok, I added the original option back, and the order is like this again.

> Similarly, the user should be able to do Shift / Ctrl clicks with the selected secondaries, to e.g. select the first and third, or deselect one, etc.

Yes, that is a good next step, but I would not mind if we keep that for a next PR and just merge the "keyboard select all option" from this PR first.

File: source/PlayerInfo.cpp [hunk 1]
```diff
@@ -2012,21 +2012,54 @@ void PlayerInfo::SelectNext()
...
+		if(it->first->Icon())
+			selectedWeapons.insert(it->first);
+	
+	// If we have only one weapon selected at this point, then the player
+	// only has a single secondary weapon. Clear the list, since the weapon
+	// was selected when we entered this function.
+	if(selectedWeapons.size() == 1)
+		selectedWeapons.clear();
+
+	return;
```
Comment (Amazinite) [inline source/PlayerInfo.cpp:2062]: No need to have a return at the end of a void function.
```suggestion
```

**Review by Amazinite — CHANGES_REQUESTED** [review]

File: source/AI.cpp [hunk 2]
```diff
@@ -3496,7 +3496,8 @@ void AI::MovePlayer(Ship &ship, const PlayerInfo &player, Command &activeCommand
 			int index = 0;
 			for(const Hardpoint &hardpoint : ship.Weapons())
 			{
-				if(hardpoint.IsReady() && hardpoint.GetOutfit() == player.SelectedWeapon())
+				const auto playerSelectedWeapons = player.SelectedWeapons();
+				if(hardpoint.IsReady() && (playerSelectedWeapons.find(hardpoint.GetOutfit()) != playerSelectedWeapons.end()))
 					command.SetFire(index);
```
[the remark this answers is not in the document]
Comment (petervdmeer) [reply source/AI.cpp]: Done

**Review by Amazinite — APPROVED** [review]

**Pull request merged**